### PR TITLE
fix(types): positive decimal from string must be finite

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1181

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: types/math/dec.go [hunk 1]
```diff
@@ -258,6 +258,10 @@ func (x Dec) IsPositive() bool {
 	return !x.dec.Negative && !x.dec.IsZero()
 }
 
+func (x Dec) IsFinite() bool {
```
Comment (aleem1314) [inline types/math/dec.go:261]: nit: godoc

**Review by aleem1314 — APPROVED** [review]

**Comment by aaronc** [comment]
Tests?

**Comment by ryanchristo** [comment]
> Tests?

We already have tests and they were intermittently failing without this change https://github.com/regen-network/regen-ledger/issues/1181

File: types/math/dec.go (hunk 1 above)
Comment (ryanchristo) [reply types/math/dec.go:265]: https://github.com/regen-network/regen-ledger/pull/1210/commits/ed80b0ab435053cc931c6f3332981787a22680a6

**Review by aaronc — COMMENTED** [review]
We don't want those .fail files I think

File: types/math/dec_test.go [hunk 2]
```diff
@@ -552,6 +553,18 @@ func testIsPositive(t *rapid.T) {
 	require.Equal(t, f > 0, dec.IsPositive())
 }
 
+// Property: isFinite(f) == isFinite(NewDecFromString(f.String()))
+func testIsFinite(t *rapid.T) {
```
[the remark this answers is not in the document]
Comment (ryanchristo) [reply types/math/dec_test.go]: https://github.com/regen-network/regen-ledger/pull/1210/commits/b83eb23583afd2c73d6b65e444498be6f0919dbf

**Pull request merged**